pilot/pkg/model/xds_cache.go

```diff
@@ -92,7 +92,7 @@ type XdsCacheEntry interface {
 	DependentTypes() []kind.Kind
 	// DependentConfigs is config items that this cache key is dependent on.
 	// Whenever these configs change, we should invalidate this cache entry.
-	DependentConfigs() []ConfigKey
+	DependentConfigs(allocator *Allocator) []ConfigKey
```

Comment:
on second thought.. could we just pass the `preAllocKeys []ConfigKey` itself here; the implementations just `append` so nil slice is okay. I think this simplifies the usage a lot.